libs/api/metadata-converter/src/lib/gn4/gn4.field.mapper.ts

```diff
@@ -286,12 +286,9 @@ export class Gn4FieldMapper {
         },
         output
       ),
-    cl_hierarchyLevel: (output, source) => {
-      const hierarchyLevel = selectField(
-        getFirstValue(selectField(source, 'cl_hierarchyLevel')),
-        'key'
-      )
-      const kind = hierarchyLevel === 'service' ? 'service' : 'dataset'
+    resourceType: (output, source) => {
+      const resourceType = selectField(source, 'resourceType')
+      const kind = resourceType[0] === 'service' ? 'service' : 'dataset'
```

Comment:
I know this sounds counter-intuitive but you should use the atomic operations here:
```ts
selectField(
  getFirstValue(selectField(source, 'resourceType')),
  'key'
)
```

This is because the documents coming from ES can sometimes have various shapes and atomic operations functions will be tolerant to nullish/unexpected values :)

Reply:
Thanks for the suggestion!